### DISCUSSION TITLE
Backup and Restore | Rule Configuration Missing

### DISCUSSION THREAD

**Discussion by SatyaaDixit** [emqx/emqx#13254] [Q&A] [answered]
Hi There,
Incase of disaster to recover the EMQX configuration was performing “Backup and Restore”.
While testing the functionality, got to know the “rule engine” configuration is missing when restore.

EMQX details : 5.1.1

Referred below link- [Backup and Restore | EMQX Docs 1](https://docs.emqx.com/en/emqx/latest/operations/backup-restore.html)

Is there some wrong in the steps I’m doing? Please support.
export: ./emqx ctl data export
import : ./emqx ctl data import

Thanks in advance.
Satya

**Comment by ieQu1** [comment]
These commands backup and restore contents of the builtin database. Rules aren't stored in the builtin database, they are part of the regular config. When you create a rule via dashboard, it's added to `data/configs/cluster.hocon`.

**Reply by SatyaaDixit** [reply]
Thanks for quick response and yes I understand the rule config there in "data/configs/cluster.hocon", and when "emqx ctl data export" is being executed this file also come along "tar.gz" package but then why did import miss it to restore?

Is there other way to do the rule import (excluding API) ? Could you help me out?

**Reply by ieQu1** [reply]
This was fixed in https://github.com/emqx/emqx/releases/tag/v5.1.2

**Reply by SergeTupchiy** [reply, marked as answer]
Hi @SatyaaDixit ,
It was a bug in 5.1.1, that didn't allow restoring rules. It was fixed in 5.1.2

**Reply by SatyaaDixit** [reply]
Thanks for the response! It helped.
